ruby_event_store-active_record/spec/migration_generator_spec.rb

```diff
@@ -0,0 +1,94 @@
+require "spec_helper"
+require_relative "../../support/helpers/silence_stdout"
+
+module RubyEventStore
+  module ActiveRecord
+    RSpec.describe MigrationGenerator do
+      around { |example| SilenceStdout.silence_stdout { example.run } }
+
+      before { allow(Time).to receive(:now).and_return(Time.new(2022, 11, 30, 21, 37, 00)) }
+
+      context "with default migration path" do
+        around do |example|
+          begin
+            @dir = FileUtils.mkdir_p("./db/migrate").first
+            example.call
+          ensure
+            FileUtils.rm_rf(@dir)
+            FileUtils.rmdir("./db")
+          end
+        end
+
+        it "is created" do
+          RubyEventStore::ActiveRecord::MigrationGenerator.new.call("binary")
+          File.read("#{@dir}/20221130213700_create_event_store_events.rb")
+        end
+      end
+
+      context "with custom directory" do
+        around do |example|
+          begin
+            @dir = Dir.mktmpdir(nil, "./")
+            example.call
+          ensure
+            FileUtils.rm_r(@dir)
+          end
+        end
+
+        subject do
+          RubyEventStore::ActiveRecord::MigrationGenerator.new.call("binary", migration_path: "#{@dir}/")
+          File.read("#{@dir}/20221130213700_create_event_store_events.rb")
+        end
+
+        it "uses particular migration version" do
+          expect(subject).to match(/ActiveRecord::Migration\[4\.2\]$/)
+        end
+
+        context "returns path to migration file" do
+          subject { RubyEventStore::ActiveRecord::MigrationGenerator.new.call("binary", migration_path: "#{@dir}/") }
+
+          it { is_expected.to match("#{@dir}/20221130213700_create_event_store_events.rb") }
+          it { is_expected.to match(File.expand_path("../#{@dir}", __dir__)) }
+        end
+
+        context "when data_type option is specified" do
```

Comment:
1. This context structure suggests that `--data-type` can only be specified for migrations within custom directory — I'm not sure why different data type variants tests are nested under "with custom directory" context.

2. Subject can be named, eg. `subject(:migration_file) { bla bla }` which can result in:

```ruby
context "with a binary datatype" do
   let(:data_type) { "binary" }
   specify { expect(migration_file).to match(/t.binary\s+:metadata/) }
end
```
I would also prefer to pass this `data_type` explicitly, but it's just me, no longer a member of rSpEc GaNg.
This test is relatively short and it's still readable, so ignore my comment, but please fix the nesting of those contexts...

Reply:
Thanks for the feedback. I got soaked into the rSpEc G4nG game. Refactored those tests. Hope they're easier to read right now.